src/commands/createFunction/openAPISteps/OpenAPICreateStep.ts

```diff
@@ -0,0 +1,116 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { OpenDialogOptions, ProgressLocation, Uri, window, workspace } from "vscode";
+import { AzureWizardExecuteStep } from "vscode-azureextensionui";
+import { ProjectLanguage } from "../../../constants";
+import { ext } from "../../../extensionVariables";
+import { localize } from "../../../localize";
+import { cpUtils } from "../../../utils/cpUtils";
+import { IFunctionWizardContext } from "../IFunctionWizardContext";
+
+export class OpenAPICreateStep extends AzureWizardExecuteStep<IFunctionWizardContext> {
+    public priority: number = 220;
+
+    public async execute(wizardContext: IFunctionWizardContext): Promise<void> {
+        const uris: Uri[] = await this.askDocument();
+        const uri: Uri = uris[0];
+        const args: string[] = [];
+
+        const errorMessage: string = 'Please make sure you have Autorest installed. Visit https://aka.ms/autorest to find more information on Autorest and installation steps.';
+
+        if (!await isNPMInstalled()) {
+            throw new Error(localize("notSupported", `NPM is not installed. ${errorMessage}`));
+        } else if (!await isAutoRestInstalled()) {
+            throw new Error(localize("notSupported", `Autorest could not be detected on the system. ${errorMessage}`));
+        }
+
+        let namespace: string = '';
+        if (wizardContext.language === ProjectLanguage.CSharp || wizardContext.language === ProjectLanguage.Java) {
+            namespace = await askNamespace(wizardContext.language);
+        }
+
+        args.push(`--input-file:${cpUtils.wrapArgInQuotes(uri.fsPath)} --version:3.0.6314`);
+
+        if (wizardContext.language === ProjectLanguage.TypeScript) {
```

Comment:
nit: I would use a 'switch case' instead of all these 'else ifs'

Reply:
Thanks. noob mistake. Coming from Python doesn't help. :)